### ISSUE TITLE
In ownCloud 8.2 GPX with the shared URL doesn't show map tiles

### ISSUE THREAD

**Issue by kayrus** [Frank1604/files_gpxviewer_extended#14]
Because of incorrect "content-security-policy" header:

```
content-security-policy: default-src 'none';script-src 'self' 'unsafe-eval';style-src 'self' 'unsafe-inline';img-src 'self' data: blob:;font-src 'self';connect-src 'self';media-src 'self';frame-src 'self'
```

When you are logged in, your browser receives these headers:

```
content-security-policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *
```

**Comment by ayonix** [comment]
I've looked into this, but you seem to only be able to change the content-security-policy in controllers/attached to a response, since it's a HTTP-header. Because this plugin is a script only, I haven't yet found a way to set it.

Also the possibility to configure it owncloud-wide through config.php was removed.

**Comment by Frank1604** [comment]
You're right, there is no way to change the content-security-policy at this time. But I'm working on a new version, which works with a display controller. I still think this week I'll finish.

**Comment by kayrus** [comment]
@Frank1604 awesome. thank you very much.

**Comment by Frank1604** [comment]
And here it is ;)

**Comment by ayonix** [comment]
Works great if the file isn't accessed over a shared link.

```
[Error] Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-eval'". (Agjbp97XQkuS6zm, line 0)

[Warning] Unexpected CSS token: : (styles.css, line 33)
[Warning] Unexpected CSS token: : (fixes.css, line 108)
[Warning] Unexpected CSS token: : (style.css, line 91)
[Error] Unrecognized Content-Security-Policy directive 'child-src'.

[Error] Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-eval'". (files_gpxviewer_extended, line 0)

[Error] Failed to load resource: the server responded with a status of 404 (HTTP/2.0 404) (en.js, line 0)
[Warning] Unexpected CSS token: : (leaflet.ie.css, line 21)
[Error] TypeError: undefined is not an object (evaluating 'FileList.setViewerMode')
```

**Comment by ayonix** [comment]
Ah. It seems it is a Safari/Upstream bug. It works in Chrome ;) Thanks!

**Comment by Frank1604** [comment]
fixed in 1.6
